askama_derive/src/input.rs

```diff
@@ -88,61 +69,76 @@ impl<'a> TemplateInput<'a> {
         match (&source, ext.is_some()) {
             (&Source::Path(_), true) => {
                 panic!("'ext' attribute cannot be used with 'path' attribute")
-            }
+            },
             (&Source::Source(_), false) => {
                 panic!("must include 'ext' attribute when using 'source' attribute")
-            }
-            _ => {}
+            },
+            _ => {},
         }
 
-        let escaping = match escaping {
-            Some(m) => m,
-            None => {
-                let ext = match source {
-                    Source::Path(ref p) => Path::new(p)
-                        .extension()
-                        .map(|s| s.to_str().unwrap())
-                        .unwrap_or(""),
-                    Source::Source(_) => ext.as_ref().unwrap(), // Already panicked if None
-                };
-                if HTML_EXTENSIONS.contains(&ext) {
-                    EscapeMode::Html
-                } else {
-                    EscapeMode::None
-                }
-            }
-        };
-
-        let parent = match ast.data {
-            syn::Data::Struct(syn::DataStruct {
-                fields: syn::Fields::Named(ref fields),
-                ..
-            }) => fields.named.iter().filter_map(|f| {
-                f.ident
-                    .as_ref()
-                    .and_then(|name| if name == "_parent" { Some(&f.ty) } else { None })
-            }),
-            _ => panic!("derive(Template) only works for struct items"),
-        }.next();
-
         let path = match source {
-            Source::Source(_) => match ext {
-                Some(ref v) => PathBuf::from(format!("{}.{}", ast.ident, v)),
-                None => PathBuf::new(),
-            },
-            Source::Path(ref s) => path::find_template_from_path(s, None),
+            Source::Source(_) => PathBuf::from(format!("{}.{}", ast.ident, ext.unwrap())),
```

Comment:
`ext` is guaranteed to be defined since it's checked above.

Reply:
Sure it is defined, but it might still be `None`. I prefer to avoid `unwrap()` on principle if possible. In this case, it might make sense to merge the `match (&source, ext.is_some())` with this `match` in a way that the `ext` can be pushed into the source.

Reply:
I think I meant to say it's guaranteed to not be `None` (to be `Some`). But your last sentence is a good idea.